File: Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> Z;
   char c;
   for (int i=0; i<s.length(); i++) {
      if (s[i]=='('||s[i]=='['||s[i]=='{') {
         Z.push(s[i]);
         continue;
      }
      if (Z.empty())
         return false;
      switch (s[i]) {
      case ')':
         c = Z.top();
         Z.pop();
         if (c=='{' || c=='[')
            return false;
         break;
      case '}':
         c = Z.top();
         Z.pop();
         if (c=='(' || c=='[')
            return false;
         break;
      case ']':
         c = Z.top();
         Z.pop();
         if (c =='(' || c == '{')
            return false;
         break;
      }
   }
   return (Z.empty());
}
        
    
};

File: valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        dic={"}":"{",")":"(","]":"["}
        for i in range(len(s)):
            if(s[i]=="(" or s[i]=="{" or s[i]=="["):
                stack.append(s[i])
            elif(s[i]==")" or s[i]=="}" or s[i]=="]"):
                if(len(stack)==0  or dic[s[i]]!=stack[-1] ) :return False 
                elif(dic[s[i]]==stack[-1]):
                    stack.pop()
                  
                
        return True if len(stack)==0 else False